### PR TITLE
Anthrogram of Shares

### LINKED ISSUE #121: Create the Share Anthrogram
```[tasklist]
### Tasks
- [x] Create the Share Anthrogram;
- [x] Add tests for it.
```

### PR DESCRIPTION
**Pull request by arnonrdp**
Created new chart to display Prompts and Entries shares.

Fix #121

### CODE REVIEW

**Comment by stravid87** [comment]
Code is clean. Component works great! I love how dynamic it is. Nicely encapsulated. I imagine a dashboard where people pick and chose what they want to see kind of like widgets / components.

There should be no side effects from this. 

I think before you merge though you got to put at least 1 test on it. A Cypress or Vitest. @hermannleboss what do you think? Should force him to add at least ONE test? haha

**Comment by arnonrdp** [comment]
> Code is clean. Component works great! I love how dynamic it is(There should be no side effects from this). Nicely encapsulated. I imagine a dashboard where people pick and chose what they want to see kind of like widgets / components.
> 
> I think before you merge though you got to put at least 1 test on it. A Cypress or Vitest. @hermannleboss what do you think? Should force him to add at least ONE test? haha

Fair and Perfect!

It is done.

**Comment by hermannleboss** [comment]
> Code is clean. Component works great! I love how dynamic it is(There should be no side effects from this). Nicely encapsulated. I imagine a dashboard where people pick and chose what they want to see kind of like widgets / components.
> 
> I think before you merge though you got to put at least 1 test on it. A Cypress or Vitest. @hermannleboss what do you think? Should force him to add at least ONE test? haha

yes i agree

**Comment by stravid87** [comment]
@arnonrdp: On it. I'll be reviewing today or tomorrow.

File: src/stores/shares.js
Comment (stravid87) [inline src/stores/shares.js:26]: ```const docId = socialNetwork + '-' + (userStore.isAuthenticated ? userStore.getUserRef.id : userStore.getUserIp)```
We should try to use the Hassh(IP) as much as possible.

**Review by stravid87 — CHANGES_REQUESTED** [review]
After much experimentation, I found a way to actually build out the tests usefully. I put a third one as an example. I couldn't figure out exactly how the interval functions worked so I left it for you. I'm gonna post to the discord channel. I think I have a pretty good idea about how to move forward with unit testing without it being SO, SO, painful every time. 

Requested changes: just at some tests to ensure the interval functions are  working.

**Comment by arnonrdp** [comment]
> Requested changes: just at some tests to ensure the interval functions are working.

~I am~ Cypress is double clicking in shares to check if Graph is visible.

Let me know your thoughts about it:

https://user-images.githubusercontent.com/26576201/227053622-b49bc0d1-907b-45f1-966c-71a577520fd7.mp4

File: test/cypress/e2e/share-prompt.cy.js
Comment (hermannleboss) [inline test/cypress/e2e/share-prompt.cy.js:20]: Can you add somme comment then we can understand what you are tring

**Comment by stravid87** [comment]
Thoughts: 
1) We def don't need to upload videos. @arnonrdp, I agree. I see it easily disabled in cypress.config.js. Alternatively, we could add a `videos/` to the `.gitignore`

**Comment by arnonrdp** [comment]
As explained by @hermannleboss, videos already ignored.
And today I installed Cypress in another project and I noticed they ignore videos by default.

File: src/components/Graphs/SharesPie.vue
Comment (stravid87) [inline src/components/Graphs/SharesPie.vue:78]: Ahhhh, I get it now: this creates instant feedback so that the use sees the share they just did.

**Review by stravid87 — COMMENTED** [review]
Well done. If you had pushed for it, I would have accepted the Vitest `expect.toBeDefined()` . The addition of the working Cypress tests (and ci/cd) is a huge bonus though so I'm glad you didn't. 

I'm ready to merge once @hermannleboss  comments on the two files that affect him ("@hermannleboss can you comment on changes to "src/components/Graphs/LikesBar.vue" & "src/utils/stats.js". Not much to say I assume. Forgive me if you've already commented on it somewhere I didn't see."). I'll DM him and then try to merge tonight before tomorrow.

File: src/components/Graphs/SharesPie.vue
Comment (arnonrdp) [reply src/components/Graphs/SharesPie.vue:78]: # 🎯

File: test/cypress/e2e/share-prompt.cy.js
Comment (hermannleboss) [inline test/cypress/e2e/share-prompt.cy.js:5]: Wait is not a good practice. 

If you need to wait someting will appear you can use  timeout instead


### Ressources 

[Wait Anti patern](https://docs.cypress.io/api/commands/wait#Time)

[Timeout](https://docs.cypress.io/guides/core-concepts/retry-ability#Timeouts)

**Review by hermannleboss — APPROVED** [review]

**Review by stravid87 — APPROVED** [review]

File: test/cypress/e2e/share-prompt.cy.js
Comment (stravid87) [reply test/cypress/e2e/share-prompt.cy.js:5]: I see that you approved changes. I'll approve now to and merge. We're all learning Cypress so we can circle round and get this anti-pattern later. Personally, Cypress is top on my list next week.

Comment (hermannleboss) [reply test/cypress/e2e/share-prompt.cy.js:5]: Yes it's why i approve.

Comment (arnonrdp) [reply test/cypress/e2e/share-prompt.cy.js:5]: OH, good to know!
Let's change it in `develop` branch then.

**Pull request merged**
